python/pyspark/sql/datasource.py

```diff
@@ -469,6 +501,200 @@ def stop(self) -> None:
         ...
 
 
+class SimpleInputPartition(InputPartition):
```

Comment:
Why do we need this in the public API? Why can't user define their own input partition class?

Reply:
Simple reader doesn't expose partitioning to user, so this is only used for the wrapper to make simple reader integrate with streaming engine. Where should we move these internal code if we don't want it to be public API?

Reply:
Yea let's not make it a public API

Reply:
I see, moved it to another file.